ffcx/codegeneration/integral_generator.py

```diff
@@ -235,85 +231,78 @@ def declare_table(self, name, table):
-        pre_definitions, parts = self.generate_partition(arraysymbol, F, "piecewise", None)
-        assert len(pre_definitions) == 0, "Quadrature independent code should have not pre-definitions"
-        parts = L.commented_code_list(
-            parts, f"Quadrature loop independent computations for quadrature rule {quadrature_rule.id()}")
-
-        return parts
+        return self.generate_partition(arraysymbol, F, "piecewise", None)
 
     def generate_varying_partition(self, quadrature_rule):
 
         # Get annotated graph of factorisation
         F = self.ir.integrand[quadrature_rule]["factorization"]
-
         arraysymbol = L.Symbol(f"sv_{quadrature_rule.id()}", dtype=L.DataType.SCALAR)
-        pre_definitions, parts = self.generate_partition(arraysymbol, F, "varying", quadrature_rule)
-        parts = L.commented_code_list(parts, f"Varying computations for quadrature rule {quadrature_rule.id()}")
-
-        return pre_definitions, parts
+        return self.generate_partition(arraysymbol, F, "varying", quadrature_rule)
 
     def generate_partition(self, symbol, F, mode, quadrature_rule):
 
-        definitions = dict()
-        pre_definitions = dict()
+        definitions = []
         intermediates = []
 
-        use_symbol_array = True
-
         for i, attr in F.nodes.items():
             if attr['status'] != mode:
                 continue
             v = attr['expression']
             mt = attr.get('mt')
 
-            # Generate code only if the expression is not already in
-            # cache
+            # Generate code only if the expression is not already in cache
             if not self.get_var(quadrature_rule, v):
                 if v._ufl_is_literal_:
                     vaccess = L.ufl_to_lnodes(v)
-                elif mt is not None:
-                    # All finite element based terminals have table
-                    # data, as well as some, but not all, of the
-                    # symbolic geometric terminals
+                elif mt:
+                    assert mt is not None
                     tabledata = attr.get('tr')
 
                     # Backend specific modified terminal translation
-                    vaccess = self.backend.access.get(mt.terminal, mt, tabledata, quadrature_rule)
-                    predef, vdef = self.backend.definitions.get(mt.terminal, mt, tabledata, quadrature_rule, vaccess)
-                    if predef:
-                        access = predef[0].symbol.name
-                        pre_definitions[str(access)] = predef
-
-                    # Store definitions of terminals in list
-                    assert isinstance(vdef, list)
-                    definitions[str(vaccess)] = vdef
+                    vaccess = self.backend.access.get(mt, tabledata, quadrature_rule)
+                    vdef = self.backend.definitions.get(mt, tabledata, quadrature_rule, vaccess)
+
+                    if isinstance(vdef, L.Section):
```

Comment:
Change to assert as it should always be true (according to the almighty Baratta)